frontend/docsettings.lua

```diff
@@ -430,12 +483,14 @@ function DocSettings:findCoverFile(doc_path)
     local location = G_reader_settings:readSetting("document_metadata_folder", "doc")
     local sidecar_dir = self:getSidecarDir(doc_path, location)
     local cover_file = findCoverFileInDir(sidecar_dir)
-    if not cover_file then
-        location = location == "doc" and "dir" or "doc"
-        sidecar_dir = self:getSidecarDir(doc_path, location)
-        cover_file = findCoverFileInDir(sidecar_dir)
+    if cover_file then return cover_file end
+    for _, mode in ipairs({"doc", "dir", "hash"}) do
```

Comment:
We have done one search already, do not repeat.
The check for `document_metadata_hash_enabled` should be added.
(Summarizing, the old code should be restored and the conditional "hash" search added.)

Reply:
maybe code previews only work if it's the same repository, not a fork..

Reply:
Updated, not sure if this is satisfactory though since I think we may sometimes need a search..